Source/WebKit/Shared/RemoteLayerTree/LayerProperties.h

```diff
@@ -163,7 +164,7 @@ struct LayerProperties {
     WebCore::FloatPoint3D anchorPoint { 0.5, 0.5, 0 };
     WebCore::FloatRect bounds;
     WebCore::FloatRect contentsRect { 0, 0, 1, 1 };
-    RemoteLayerBackingStoreOrProperties backingStoreOrProperties;
+    Variant<std::monostate, RemoteLayerBackingStoreOrProperties, std::unique_ptr<RemoteLayerContents>> backingStoreOrContents;
```

Comment:
While the root cause is not originating in this change, I think this change now exerbates the issue to warrant a concern:

The type now becomes variant inside variant. This is inconvenient.
The type now has multiple ways to end up with disengaged: monostate, RemoteLayerBackingStoreOrProperties both unique_ptrs nullptr, RemoteLayerContents nullptr.

Reply:
Strawman:
Rename RemoteLayerBackingStoreOrProperties to LayerPropertiesBackingStore.

```
using LayerPropertiesBackingStore = Variant<std::monostate, 
// Used in WebContentProcess.
UniqueRef<RemoteLayerBackingStore>, 
UniqueRef<RemoteLayerDelegatedBackingStore>,
// Used in UI process.
UniqueRef<RemoteLayerBackingStoreProperties>, 
UniqueRef<RemoteLayerDelegatedBackingStoreProperties>>;
```

Reply:
The crux of the underlying blocking issue issue for me seems to be that we do not have a formalism where we would be able to hold Ta on the send side and receive Tb on the receive side. This has been a constant source of confusion. 
In this case, the problem is that we encode map of changed layers WebKit::ChangedLayers, thus we need to have the domain object RemoteLayerBackingStore in the LayerProperties.

However, we don't want RemoteLayerBackingStore in the receive side, while we still want to use  LayerProperties. Or, rather, we want to keep the encode and decode in sync with all the multiple properties LayerProperties has.